tests/test_report_manager.py

```diff
@@ -55,98 +87,112 @@ def setUp(self):
+            self.model_config["name"] = f"test_model2_report_{i}"
+            model_config = ModelConfig.create_from_dictionary(self.model_config)
+            run_config = RunConfig("test_model2", model_config,
+                                   self.perf_config)
+
+            avg_non_gpu_metrics = [
+                PerfThroughput(200 + 10 * i),
+                PerfLatency(4000),
+                CPUUsedRAM(1000)
+            ]
+            measurement = Measurement(gpu_data=avg_gpu_metrics,
+                                      non_gpu_data=avg_non_gpu_metrics,
+                                      perf_config=self.perf_config)
+            measurement.set_result_comparator(result_comparator)
+
+            self.result_manager.add_measurement(run_config, measurement)
+
+        self.result_manager.compile_and_sort_results()
+        with unittest.mock.patch(
+                "model_analyzer.reports.pdf_report.open",
+                unittest.mock.mock_open(
+                    read_data=bytes(">:(".encode("ascii")))):
+            self.report_manager.create_summaries()
 
         self.assertEqual(self.report_manager.report_keys(),
-                         ['test_report1', 'test_report2'])
-        report1_data = self.report_manager.data('test_report1')
-        report2_data = self.report_manager.data('test_report2')
+                         ["test_model1", "test_model2"])
+        report1_data = self.report_manager.data("test_model1")
+        report2_data = self.report_manager.data("test_model2")
 
         self.assertEqual(len(report1_data), 10)
         self.assertEqual(len(report2_data), 5)
 
     def test_build_summary_table(self):
-        mock_model_config = MockModelConfig()
-        mock_model_config.start()
-        objective_spec = {'perf_throughput': 10}
-        self.result_comparator = ResultComparator(
-            metric_objectives=objective_spec)
+        self._setUp_managers()
+        result_comparator = ResultComparator(
+            metric_objectives={"perf_throughput": 10})
 
-        avg_gpu_metrics = {0: {'gpu_used_memory': 6000, 'gpu_utilization': 60}}
+        avg_gpu_metrics = {0: [GPUUsedMemory(6000), GPUUtilization(60)]}
 
         for i in range(10, 0, -1):
-            avg_non_gpu_metrics = {
-                'perf_throughput': 100 + 10 * i,
-                'perf_latency': 4000,
-                'cpu_used_ram': 1000
-            }
-            self.model_config['name'] = f'model_{i}'
-            model_config = ModelConfig.create_from_dictionary(
-                self.model_config)
-            self.report_manager.add_result(
-                report_key='test_report',
-                result=construct_result(
-                    avg_gpu_metric_values=avg_gpu_metrics,
-                    avg_non_gpu_metric_values=avg_non_gpu_metrics,
-                    comparator=self.result_comparator,
-                    model_config=model_config))
+            self.model_config["name"] = f"model_{i}"
```

Comment:
Even if you cannot use the `test_utils.py` functions, there seems to be a lot of repeated code here, perhaps you can refactor the measurement creation/addition into a single function.

Reply:
`test_utils.py` function will be called